x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_text_property.ts

```diff
@@ -9,16 +9,29 @@ import type { Map as MbMap } from '@kbn/mapbox-gl';
 import { DynamicStyleProperty } from './dynamic_style_property';
 import { LabelDynamicOptions } from '../../../../../common/descriptor_types';
 import { RawValue } from '../../../../../common/constants';
+import { IESAggField } from '../../../fields/agg';
 
 export class DynamicTextProperty extends DynamicStyleProperty<LabelDynamicOptions> {
   syncTextFieldWithMb(mbLayerId: string, mbMap: MbMap) {
     if (this._field && this._field.isValid()) {
       const targetName = this.getMbPropertyName();
-      mbMap.setLayoutProperty(mbLayerId, 'text-field', [
-        'coalesce',
-        [this.getMbLookupFunction(), targetName],
-        '',
-      ]);
+
+      if (
+        typeof (this._field as IESAggField).isCount === 'function' &&
```

Comment:
imho, if we are going to introduce an `isCount`-interface method anyway, I would introduce it on the `IField` directly (and not IESAggField), and have it return false for all other instances.

That will remove the requirement for the `isCount`-duck-type check. This duck-type specifically is also a little handwavy, as it relies on the cast to IESAggField to actually fail, for `isCount` to be undefined. It seems we're fighting more with the typesystem than we're actually gaining.